### ISSUE TITLE
[Menu - Grouping with heading] Heading accessibility for assistive technology

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2911]
### Provide a general summary of the issue here

The grouping with heading has a visible text indicating what the different groups in the menu are. However, the group header is not accessible to assistive technology users. Currently when using a keyboard to use the group, the users focus moves through the menu items. As the headings themselves aren't focusable this way (and definitely shouldn't be) when using keyboard, some screen reader users aren't aware of the heading/grouping existing at all. 

### 🤔 Expected Behavior?

When user moves through the list using keyboard only with screen reader, user is aware that there are groups of elements.

### 😯 Current Behavior

User does not get notified about grouping when using a keyboard.

### 💁 Possible Solution

Add role="group" and an aria-label to the grouping container.

<img width="787" height="184" alt="role and aria-label added and highlighted in Google dev tools" src="https://github.com/user-attachments/assets/f10f77d6-82a6-4b30-afcc-b8e9c5b039b3" />

### 🖥️ Steps to Reproduce

Screen reader (NVDA):
1. Open a menu with heading using keyboard
2. Go through the menu with arrow keys

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning
